From a478a557087df7886f09dc9e48ccac57bb83e06b Mon Sep 17 00:00:00 2001
From: Amit Kumar Mahapatra <amit.kumar-mahapatra@xilinx.com>
Date: Thu, 2 Apr 2020 00:07:03 +0530
Subject: [PATCH 1310/1852] spi: spi-zynqmp-gqspi: Update driver to write
 appropriate dummy cycles as per the tx_buswidth

commit ab5e7672ae439ba4fda5a47b1cbed44f1469a54c from
https://github.com/Xilinx/linux-xlnx.git

The driver was not considering the tx_buswidth while updating the GenFifo
with number of dummy cycles.
This works fine when the tx_buswidth = 1 but when tx_buswidth = 4 the
driver updates GenFifo with invalid number of dummy cycles.
This results in data corruption.
This patch fixes the issue and computes the right number of dummy cycles
by taking into account tx_buswidth while calculating the number of dummy
cycles.

Signed-off-by: Amit Kumar Mahapatra <amit.kumar-mahapatra@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/spi/spi-zynqmp-gqspi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/spi/spi-zynqmp-gqspi.c b/drivers/spi/spi-zynqmp-gqspi.c
index 44f605372a28..27c793cdf634 100644
--- a/drivers/spi/spi-zynqmp-gqspi.c
+++ b/drivers/spi/spi-zynqmp-gqspi.c
@@ -720,7 +720,7 @@ static void zynqmp_qspi_preparedummy(struct zynqmp_qspi *xqspi,
 	*genfifoentry &= ~GQSPI_GENFIFO_IMM_DATA_MASK;
 
 	if (transfer->dummy)
-		*genfifoentry |= transfer->dummy;
+		*genfifoentry |= (transfer->dummy / transfer->tx_nbits);
 }
 
 /**
-- 
2.31.1

